packages/core/src/types.ts

```diff
@@ -313,6 +313,39 @@ export type Postprocessor = (util: UtilObject) => void
 export type ThemeExtender<T> = (theme: T) => T | void
 
 export interface ConfigBase<Theme extends {} = {}> {
+  content?: {
+    /**
+     * Glob patterns to extract th classes from.
+     *
+     * In Vite plugin, this patterns only used to extract classes
+     * from the files that is not part of the build pipeline.
+     *
+     * In PostCSS plugin, files to be scanned are determined by
+     * the glob patterns defined in this option.
+     */
+    filesystem?: string[]
+
+    /**
+     * Plain text to be extracted
+     */
+    plain?: (string | { content: string; extension: string })[]
```

Comment:
I wouldn't suggest doing that. It would be confusing in many ways. Let's say we have 2 plain content objects with `id` property. What `id` should be? A path, relative or absolute, should it exist in fs or just an extension? And most importantly `id`s should be unique, This will force people to think of a unique name for their `id`s. We somewhat forcing them to overthink. `extension` name clearly shows all we need is an extension.

Reply:
It's sometimes about extensions. Extractors can check the module id and apply conditionally based on it (under certain folders, or conventions). For example, in Vue we will have

- `foo.vue`
- `foo.vue?vue&type=template&lang=pug`
- `foo.vue?vue&type=style&lang=css`

If I want to apply transformation before passing the extractor, I will need to compile the pug template to HTML, in this case, `extension` can be limited.

Maybe this would not be used often, but being able to pass the full path (it's always named as `id` in other places) would make it extendable in the future and more consistent across the pipeline.

If you are concerned about the understanding, I think we could document to explain it in jsdocs or the docs site.